resim/msg/converter_plugin.cc

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 ReSim, Inc.
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+#include "resim/msg/converter_plugin.hh"
+
+#include <dlfcn.h>
+
+#include <utility>
+
+#include "resim/assert/assert.hh"
+
+namespace resim::msg {
+
+ConverterPlugin::ConverterPlugin(const std::filesystem::path &plugin_path)
+    : handle_(dlopen(plugin_path.c_str(), RTLD_LAZY)) {
+  REASSERT(handle_ != nullptr, "Failed to load plugin!");
+
+  converter_ =
+      // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+      reinterpret_cast<const Converter>(dlsym(handle_, CONVERTER_NAME));
+  REASSERT(converter_ != nullptr, "Failed to load_plugin!");
+}
+
+ConverterPlugin::~ConverterPlugin() { dlclose(handle_); }
+
+std::optional<rclcpp::SerializedMessage> ConverterPlugin::try_convert(
+    const std::string_view ros2_message_type,
+    const rclcpp::SerializedMessage &ros2_message) const {
```

Comment:
I would be tempted to add a `if (converter == nullptr)` check here. In theory, should be caught during construction, but given it's a dynamically loaded function, not a bad error to check...

Reply:
Okay. I've added this check and similar for the other functions I added.